### DISCUSSION TITLE
NetBox v2.11 beta release & demo server

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#6093] [General]
I'm excited to announce that the first beta release for NetBox 2.11 is [now available](https://github.com/netbox-community/netbox/releases/tag/v2.11-beta1)! This is arguably our most ambitious release yet, packed with new features including:

* Object journaling support (#151)
* Parent interface assignments (#1519)
* Pre-/post-change data in change logs & webhooks (#3451, #5913)
* Marking interfaces as connected without a cable (#3648)
* Assigning devices to locations within a site (#4971)
* Dynamic object exports (#4999)
* Improved VLAN group scoping (#5284)
* Extended support for custom fields (#5370, #5401)
* REST API endpoints for webhooks & custom links (#5608, #5610)
* Site groups (#5892)
* Provider networks (#5986)

...and plenty more! See the [release notes](https://netbox.readthedocs.io/en/feature/release-notes/version-2.11/) for a complete rundown.

To help with testing the v2.11 beta, we've launched a public demo site at https://beta.netbox.dev/. Log in with the username `admin` and password `admin` to check it out. The database will be restored from a snapshot automatically every day at 7:00am UTC.

As with every beta release, your feedback is crucial to ensuring the stability and completeness of the final product. Please be sure to [report any suspected bugs](https://github.com/netbox-community/netbox/issues/new/choose), and let us know if the release meets your expectations.

**Comment by andershagman** [comment]
Great job!

I have concentrated on the location feature. Here are some thoughts: 
In the site view with, as we have, 20 buildings, 9 floors and 1 or 2 rack rooms per floor, it's a long list.  
![Bild 1](https://user-images.githubusercontent.com/20284974/113911745-9b5a8e80-97da-11eb-9780-8f50d792e15c.png)

You can not have floors with the same name in the same site. We have floor 1-9 in 20 buildings. 
![Bild 2](https://user-images.githubusercontent.com/20284974/113912432-6569da00-97db-11eb-8e54-5c4f8b5fd564.png)

If you search on room 13 I get nothing.
![bild3 ](https://user-images.githubusercontent.com/20284974/113912656-b083ed00-97db-11eb-8520-3716ca3cbbff.png)

If you search on room 13 in site DM-Akron I get the racked devices.
![Bild 4](https://user-images.githubusercontent.com/20284974/113912941-03f63b00-97dc-11eb-9f02-b4cd441d4008.png)

**Reply by jeremystretch** [reply]
Honestly it looks like you'd be better served by defining your buildings as sites.

**Reply by dreng** [reply]
You might come accross the same problem with other types of locations. Let's say you've got two production halls, each of them has got a location for quality department. That's two different sites with one name for two different locations. I tend to say that a child must be unique in the scope of it's parent only.

I would agree though that the naming convention is not ideal. I'd suggest something like 1.2.7 for building 1, second floor, room 7. Thus "Building 03, Floor 4" becomes 3.4. While it would be preferable that netbox handles this issue I think it should be o.k. to develop a suitable naming convention.

**Reply by andershagman** [reply]
As Dreng said:
"I tend to say that a child must be unique in the scope of it's parent only"
If we get rid of the slug and the name does not have to be uniq, we have the id.
If I can avoid splitting one site into many. For me a site is an organizational unit with address, contact, telephone and so on. An area with one or more buildings.

The problem is other management software have there way to divide the world. We are in the works to implement SD-A and there you have area, building and floor. We also have ACI and a CMDB with there shortcomings. To bridge all this together with Netbox I would really like Netbox to have the most flexible way to handle this.

Yes I have played with chained names.
site     building      floor      room
lsk      lsk-18         18-02    02-123

**Reply by jeremystretch** [reply]
> Let's say you've got two production halls, each of them has got a location for quality department. That's two different sites with one name for two different locations.

Right, and that's supported: You can have a "Quality department" location defined within each site.

@andershagman I'm sorry but I don't follow what you're saying.

**Comment by dreng** [comment]
Well done Jeremy!

I just had a sneak peek at the demo and tested some things regarding the locations. Things I would suggest to improve so far:
- Locations at the organization panel on the home page would be nice
- Add a stats panel like it is implemented at the sites view (locations page)
- Add the ability to show not only devices of the current location but all devices in all child locations as well (locations page)
- Add the ability to filter by location with children (devices page)

And I found a small bug when adding a new device: It is possible to choose inconsistent values within site and location. If you choose the site first you'll notice that the location drop down widget filters possible values because locations belong to specific sites. But if you choose the location first instead then the sites are not filtered. It is possible to choose any site and save the new device which leads to an inconsistent state.

**Reply by jeremystretch** [reply]
Please be sure to submit feature requests for any improvements you'd like to see!

**Reply by jcralbino** [reply]
Hello @jeremystretch 

I do see great enhancements on 2.11 , I do like the change to locations as it seems more clear than rack groups. 

Can you tell me when is expected the release of the final version of 2.11?

**Reply by jeremystretch** [reply]
Probably in around 30 minutes. :slightly_smiling_face:

**Reply by dreng** [reply]
Just like Duke Nukem 4ever: When it's done ;-)

@jeremystretch Would you kindly have a look at #6124 again? I'm not sure if you get notified when someone commented on a closed topic. The change you made there does not what it was supposed to do IMO.

**Comment by candlerb** [comment]
In the release notes under [REST API changes](https://netbox.readthedocs.io/en/feature/release-notes/version-2.11/#rest-api-changes), there appears to be a formatting problem, e.g. the third bullet point just says:

* All device components

I think the intention was for nested bullet points:

* All device components
    * Added support for custom fields
    * Added created and last_updated fields to track object creation and modification
* All device component templates
    * Added created and last_updated fields to track object creation and modification
* All organizational models
    * Added support for custom fields
... etc

You could try indenting the child bullets by 4 spaces instead of 2 spaces (that works for me with Markdown)

**Reply by jeremystretch** [reply]
> You could try indenting the child bullets by 4 spaces instead of 2 spaces

Yep, that was it. Thanks!
